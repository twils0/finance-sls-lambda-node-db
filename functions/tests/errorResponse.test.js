const { errorResponse } = require('../errorResponse');

global.console.log = jest.fn();

const stringError = 'not an object';
const noCodeError = {
  message: 'testMessage',
};
const codeError = {
  code: 'testCode',
  message: 'testMessage',
};

const res = {
  status: 400,
  body: {
    error: codeError,
  },
};
const expectedResult = `[400] ${JSON.stringify(res)}`;

describe('functions', () => {
  describe('errorResponse', () => {
    afterEach(() => {
      global.console.log.mockReset();
    });

    it('returns callback error, when provided a string error', async () => {
      const result = errorResponse(stringError);

      expect(result).toMatchSnapshot();
      expect(global.console.log).toMatchSnapshot();
    });

    it("returns an unexpected error, when provided an object error without a 'code' key", async () => {
      const result = errorResponse(noCodeError);

      expect(result).toMatchSnapshot();
      expect(global.console.log).toMatchSnapshot();
    });

    it("returns callback error, when provided an object error with a 'code' key", async () => {
      const result = errorResponse(codeError);

      expect(result).toEqual(expectedResult);
      expect(global.console.log).toMatchSnapshot();
    });
  });
});
