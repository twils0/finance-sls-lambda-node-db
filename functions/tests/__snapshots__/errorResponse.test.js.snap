// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions errorResponse returns an unexpected error, when provided an object error without a 'code' key 1`] = `"[400] {\\"status\\":400,\\"body\\":{\\"error\\":{\\"code\\":\\"unexpected_error\\",\\"message\\":\\"Please contact customer service.\\"}}}"`;

exports[`functions errorResponse returns an unexpected error, when provided an object error without a 'code' key 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "---ERR0R---
",
      "{\\"message\\":\\"testMessage\\"}",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`functions errorResponse returns callback error, when provided a string error 1`] = `"[400] {\\"status\\":400,\\"body\\":{\\"error\\":{\\"code\\":\\"unexpected_error\\",\\"message\\":\\"Please contact customer service.\\"}}}"`;

exports[`functions errorResponse returns callback error, when provided a string error 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "---ERR0R---
",
      "\\"not an object\\"",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`functions errorResponse returns callback error, when provided an object error with a 'code' key 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "---ERR0R---
",
      "{\\"code\\":\\"testCode\\",\\"message\\":\\"testMessage\\"}",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
